### PR TITLE
Play mode switching fix

### PR DESCRIPTION
**Pull request by amirebrahimi**

### CODE REVIEW

**Comment by mtschoen-unity** [comment]
What does this fix, exactly? From what I see, it just moves the handling of the editor pref into the context manager?

Also, since we have this and saving of layout, should we add a play button to the menu?

**Comment by amirebrahimi** [comment]
It moves the methods over, but the main thing is that EditorApplication.playmodeStateChanged was not getting called. In re: to adding a play button - that relies on having an action menu, which doesn't exist right now. As a stop-gap we could add play to the radial menu, since it is an action.

**Comment by mtschoen-unity** [comment]
Got it. Looks good to me.

I don't think play makes much sense on the radial menu, so let's hold off.

File: Scripts/Core/Contexts/EditingContextManager.cs [hunk 1]
```diff
@@ -89,6 +94,32 @@ static void EditProjectSettings()
 			Selection.activeObject = settings;
 		}
 
+		// Life cycle management across playmode switches is an odd beast indeed, and there is a need to reliably relaunch
+		// EditorVR after we switch back out of playmode (assuming the view was visible before a playmode switch). So,
+		// we watch until playmode is done and then relaunch.  
+		static void ReopenOnExitPlaymode()
+		{
+			bool launch = EditorPrefs.GetBool(k_LaunchOnExitPlaymode, false);
```
Comment (mtschoen-unity) [inline Scripts/Core/Contexts/EditingContextManager.cs]: var

File: Scripts/Core/Contexts/EditingContextManager.cs [hunk 2]
```diff
@@ -89,6 +94,32 @@ static void EditProjectSettings()
...
+					EditorApplication.delayCall += ShowEditorVR;
+			}
+		}
+
+		private void OnPlaymodeStateChanged()
+		{
+			if (EditorApplication.isPlayingOrWillChangePlaymode)
+			{
+				EditorPrefs.SetBool(k_LaunchOnExitPlaymode, true);
+				VRView view = VRView.activeView;
```
Comment (mtschoen-unity) [inline Scripts/Core/Contexts/EditingContextManager.cs]: var

**Comment by amirebrahimi** [comment]
@mtschoen-unity - changes are up

**Pull request merged**